Libraries/LMCommon/Evaluate.swift

```diff
@@ -159,62 +190,108 @@ struct RepetitionContext: Sendable {
 ///
 /// Note: this uses `asyncEval()` and there may be an async evaluation running after a call to `next()`.
 public struct TokenIterator: Sequence, IteratorProtocol {
-    let model: LLMModel
-    let parameters: GenerateParameters
+    let model: any LanguageModel
+    var state: LMOutput.State?
 
-    var y: MLXArray
+    var y: LMInput.Text
     var cache: [KVCache]
-    var repetitionContext: RepetitionContext
-    let sampleContext: SampleContext
+    var processor: LogitProcessor?
+    let sampler: LogitSampler
 
-    public init(prompt: MLXArray, model: LLMModel, parameters: GenerateParameters) {
+    public init(
+        prompt: MLXArray, model: any LanguageModel, cache: [KVCache]? = nil,
+        parameters: GenerateParameters
+    ) throws {
```

Comment:
This now takes either a prompt (MLXArray) or an LMInput (text + image + ...) via multiple initializers.